### ISSUE TITLE
org-agenda seems to be broken

### ISSUE THREAD

**Issue by sutram** [hlissner/doom-emacs#4862]
This is with the latest Doom - Doom v2.0.9 (grafted, HEAD -> develop, origin/develop, origin/HEAD 2d140a7 2021-04-14 17:21:47 -0400)

**What did you expect to happen?**

Expected SPC o A a to show org agenda

**What actually happened?**
Agenda display is blank and I get a stack trace
```
Debugger entered--Lisp error: (invalid-function (date date))
  (date date)()
  org-agenda-get-day-entries("/home/mahesh/Dropbox/Org/Getting Started with Orgz..." (4 11 2021) :deadline :scheduled :timestamp :sexp)
  apply(org-agenda-get-day-entries "/home/mahesh/Dropbox/Org/Getting Started with Orgz..." (4 11 2021) (:deadline :scheduled :timestamp :sexp))
  org-agenda-list(nil)
  funcall-interactively(org-agenda-list nil)
  call-interactively(org-agenda-list)
  org-agenda(nil)
  funcall-interactively(org-agenda nil)
  call-interactively(org-agenda nil nil)
  command-execute(org-agenda)
```

**Additional details:**
There was no problem before the Doom update today

![Screenshot from 2021-04-14 18-31-41](https://user-images.githubusercontent.com/8870300/114797127-b6378080-9d4f-11eb-8645-47ff35eb70f0.png)

**Steps to reproduce:**
1. Once Doom Emacs starts, type SPC o A a
2. The screenshot above is displayed

**System information:**
https://pastebin.com/JHYGUKw1

**Comment by hlissner** [comment]
I can't reproduce this and is likely a local byte-compilation issue. Run `doom sync && doom build` to recompile your packages.

**Comment by sutram** [comment]
@hlissner It was `doom build` that fixed it. I re-installed doom emacs, used `doom sync` but I forgot about `doom build`

I feel a bit embarrassed.

By the way, I truly appreciate doom emacs. Thanks.

**Comment by ndunn990** [comment]
### Summary
Tl;dr -> For posterity, if the above solutions don't appear to be solving your problem or seem to be producing yet more problems, this might help. Basically, the idea is to reinstall the Doom org module specifically. Remove the Doom org module from your init.el and run `doom build && doom sync`. Once complete, verify that you can now open your agenda in Emacs (keep in mind evil keybindings won't be present). If it works, reinstall the Doom org module by re-adding it to your init.el and running `doom build && doom sync`.

### Background (If You Care)

I'm weird and I'm running openSUSE Tumbleweed, so this could be, in part, the cause for my unique problems. I run openSUSE Tumbleweed on both a desktop and laptop and experienced this issue on both, so it's the most obvious common denominator.

None of these solutions helped me. I tried to `doom build && doom sync` more than once. I added in a `doom clean` at one point to see if that would help - it did not. I also tried reinstalling Doom completely, but it _still_ reported the same issue. Other times, the `doom build` would then begin to produce another error (https://github.com/hlissner/doom-emacs/issues/4171). I could eventually fix that problem only to be greeted by this org-agenda issue again.

Finally, I tried commenting out the Doom 'org' module. Full disclosure, I have more than just the org module - `(org +roam +journal)`. I then ran `doom build && doom sync` once more. I was finally able to pull up my agenda, though none of the Evil keybindings were present, of course. I then uncommented the Doom org module in my init.el and ran `doom build && doom sync` one final time. _That_ seems to have resolved my issue.

I hope this helps someone or I hope that I'm the only weird one, here.

**Comment by jrbrodie77** [comment]
@ndunn990 That was super helpful.  I just had a panic attack, no agenda, what!! Went down a debugging rabbithole but your advice did the trick. I had to remove org and vterm from my init followed by doom sync and build. Back to working like a charm.

@hlissner Thanks for Doom! It's seriously good software.

**Comment by cnoceda** [comment]
Hi! 
I've the same problem doing a reinstallation (deleting all the .emacs.d) and doing the build and sync. When I disable the org module everithing works (without org :-)) and after the org enable I get the Error [#4171](https://github.com/hlissner/doom-emacs/issues/4171) when I apply the error workaround the error [#4998](https://github.com/hlissner/doom-emacs/issues/4998) 

So, I'm in an infinite loop :-)
